BookingsAPI/Bookings.Domain/Questionnaire.cs

```diff
@@ -0,0 +1,55 @@
+﻿using Bookings.Domain.Ddd;
+using Bookings.Domain.Participants;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Bookings.Domain
+{
+    public class Questionnaire : Entity<long>
+    {
+        public Questionnaire()
+        {
+            SuitabilityAnswers = new List<SuitabilityAnswer>();
+            UpdatedDate = DateTime.UtcNow;
+        }
+
+        public Guid ParticipantId { get; set; }
+        public DateTime UpdatedDate { get; set; }
+
+        public virtual Participant Participant { get; set; }
+        public virtual IList<SuitabilityAnswer> SuitabilityAnswers { get; protected set; }
+
+        public void AddSuitabilityAnswers(IList<SuitabilityAnswer> suitabilityAnswers)
+        {
+            foreach (var suitabilityAnswer in suitabilityAnswers)
+            {
+                AddSuitabilityAnswer(suitabilityAnswer.Key, suitabilityAnswer.Data, suitabilityAnswer.ExtendedData);
+            }
+        }
+
+        public virtual void AddSuitabilityAnswer(string key, string data, string extendedData)
+        {
+            var existingSuitabilityAnswer = SuitabilityAnswers.FirstOrDefault(answer => answer.Key == key);
```

Comment:
You can null coalesce here 
SuitabilityAnswers.FirstOrDefault(answer => answer.Key == key) ??  new SuitabilityAnswer(key, data, extendedData)
                { ....

Reply:
Not in this case